service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -125,84 +120,99 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
-					Namespace: infrastructureRef.Namespace,
+				err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{
+					Namespace: cr.Namespace,
 					Name:      infrastructureRef.Name,
 				}, gcpCluster)
 				if err != nil {
 					return nil, microerror.Mask(err)
 				}
 
-				gcpProject, _, err := unstructured.NestedString(gcpCluster.Object, []string{"spec", "project"}...)
+				gcpProject, _, err = unstructured.NestedString(gcpCluster.Object, []string{"spec", "project"}...)
 				if err != nil {
 					return nil, microerror.Mask(err)
 				}
-				clusterValues["gcpProject"] = gcpProject
-
-				region, _, err := unstructured.NestedString(gcpCluster.Object, []string{"spec", "region"}...)
-				if err != nil {
-					return nil, microerror.Mask(err)
-				}
-				clusterValues["region"] = region
-
-			case "OpenStackCluster":
-				clusterValues["provider"] = "openstack"
-
-			case "VSphereCluster":
-				clusterValues["provider"] = "vsphere"
-
 			default:
-				r.logger.Debugf(ctx, "unable to extract infrastructure provider-specific clusterValues for cluster. Unsupported infrastructure kind %q", infrastructureRef.Kind)
+				r.logger.Debugf(ctx, "unable to extract infrastructure provider-specific clusterValues for cluster. Unsupported infrastructure kind %q", r.provider)
 			}
 		}
 	}
 
-	configMapSpecs := []configMapSpec{
-		{
-			Name:      key.AppOperatorValuesResourceName(&cr),
-			Namespace: cr.GetNamespace(),
-			Values:    appOperatorValues,
+	appOperatorValues := map[string]interface{}{
+		"app": map[string]interface{}{
+			"watchNamespace":    cr.GetNamespace(),
+			"workloadClusterID": key.ClusterID(&cr),
 		},
-		{
-			Name:      key.ClusterValuesResourceName(&cr),
-			Namespace: cr.GetNamespace(),
-			Values:    clusterValues,
+		"provider": map[string]interface{}{
+			"kind": r.provider,
+		},
+		"registry": map[string]interface{}{
+			"domain": r.registryDomain,
 		},
 	}
 
-	for _, spec := range configMapSpecs {
-		configMap, err := newConfigMap(cr, spec)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+	appValuesYaml, err := yaml.Marshal(appOperatorValues)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
-		configMaps = append(configMaps, configMap)
+	appValuesConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      key.AppOperatorValuesResourceName(&cr),
+			Namespace: cr.GetNamespace(),
+			Annotations: map[string]string{
+				annotation.Notes: fmt.Sprintf("DO NOT EDIT. Values managed by %s.", project.Name()),
+			},
+			Labels: map[string]string{
+				label.Cluster:   key.ClusterID(&cr),
+				label.ManagedBy: project.Name(),
+			},
+		},
+		Data: map[string]string{
+			"values": string(appValuesYaml),
+		},
 	}
 
-	return configMaps, nil
-}
+	clusterValues := ClusterValuesConfig{
+		BaseDomain:    key.BaseDomain(&cr, r.baseDomain),
+		ChartOperator: ChartOperatorConfig{Cni: map[string]bool{"install": true}},
+		Cluster: ClusterConfig{
+			Calico: map[string]string{"CIDR": podCIDR},
```

Comment:
can you add a comment why/when `Calico` is relevant. As new clusters are using `cilium` as CNI it's a bit confusing (to me :see_no_evil: ) to see this value in CAPI related clusters.

Reply:
Yeah, I actually don't know :smile: 

I did a quick search and found some apps relying on it. For example `coredns-app` itself https://github.com/giantswarm/coredns-app/blob/master/helm/coredns-app/templates/configmap.yaml#L17

So I'd keep it for now?

Reply:
good catch -  the generated `Corefile` in CAPO currently looks like this. Will fix CAPO clusters to set a proper `Calico` value

```
  Corefile: |                                                                                                                                                        
    .:1053 {                                                                                                                                                         
        cache 30                                                                                                                                                     
        errors
        health
        ready
        kubernetes cluster.local 10.96.0.0/12 10.1.0.0/16/16 {
          fallthrough in-addr.arpa ip6.arpa
          pods verified
        }
        log . {
          class denial
          class error        }        loadbalance round_robin
        prometheus :9153
        forward . /etc/resolv.conf
        reload
    }
```